docs/pages/setup/guides/teleport-operator.mdx

```diff
@@ -0,0 +1,213 @@
+---
+title: Teleport Kubernetes Operator (Preview)
+description: Easily manage Teleport resources from Kubernetes
+---
+
+<Admonition type="warning" title="Preview">
+  Teleport Kubernetes Operator is currently in Preview mode.
+</Admonition>
+
+The Teleport Kubernetes Operator provides a way for Kubernetes users to manage some Teleport resources through
+Kubernetes, following [the Operator Pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).
+
+The Teleport Kubernetes Operator is deployed alongside its CustomResourceDefinitions. Once deployed, users
+can use a Kubernetes client like `kubectl` or their existing CI/CD Kubernetes pipelines to create Teleport
+CustomResources. The Teleport Kubernetes Operator watches for those resources and does API calls to Teleport to
+reach the desired state.
+
+Currently supported Teleport resources are:
+- users
+- roles
+
+This guide covers how to:
+- Install Teleport Operator on Kubernetes.
+- Configure Teleport users and roles using Kubernetes.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- Kubernetes cluster (with or without `teleport-cluster` already deployed);
+- [Helm](https://helm.sh/docs/intro/quickstart/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/)
+
+Validate Kubernetes connectivity by running the following command:
+
+```code
+$ kubectl cluster-info
+# Kubernetes control plane is running at https://127.0.0.1:6443
+# CoreDNS is running at https://127.0.0.1:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+# Metrics-server is running at https://127.0.0.1:6443/api/v1/namespaces/kube-system/services/https:metrics-server:https/proxy
+```
+
+<Admonition type="tip">
+  Users wanting to experiment locally with the operator can use [minikube](https://minikube.sigs.k8s.io/docs/start/)
+  to start a local Kubernetes cluster:
+  ```code
+  $ minikube start
+  ```
+</Admonition>
+
+## Step 1/3. Install teleport-cluster with the operator
+
+Add the Teleport Helm Chart repo:
+```code
+$ helm repo add teleport https://charts.releases.teleport.dev
```

Comment:
Nit: May wanna instruct users to run `helm repo update` also in case they already have the repo.